src/run.js

```diff
@@ -185,7 +187,7 @@ const processPage = ({
           return safeReject(
             new Error(`${response.status()} on ${responseUrl}`)
           );
-        } else if (response.status() >= 300) {
+        } else if (response.status() >= 300 && response.status() !== 304) {
```

Comment:
Can you move this second if statement part *inside*. So it's like this:
```javascript
} else if (response.status() >= 300) {
  // 304 Not Modified is an exception to the 3xx responses.
  // It's expected to not be a redirect and not have content. 
  if (response.status() !== 304) {
    ...existing code...
  }
}
```
Otherwise, you'll end up in the `} else if (resourceType === 'stylesheet') {` around line 199. 

Look at this screenshot from Firefox's Dev tools. 
<img width="1507" alt="screen shot 2018-03-20 at 7 34 49 am" src="https://user-images.githubusercontent.com/26739/37652229-3cb9b00a-2c11-11e8-8e3c-68ae82e8d06a.png">
It's smart enough to figure out that the response type is "stylesheet" but if that was to happen in `minimalcss` it would fail unpredictably on the line `response.text().then(text => {`.
Basically for 304 Not Modified we don't want to write down their redirects *or* try to read their content.

Reply:
But for 304 we should read content. Server didn't send content - yes, but browser should have this content, so `response.text()` should work. Unless I'm confusing something - we should treat 304 as 200

Reply:
A HA! That makes sense. You're right. Arguably we should have the content already if the request was made such that the server responded with a 304. But perhaps if you pass in a `puppeteer` browser instance that already has some stuff cached, it could happen. 
I rest my case.